src/algo_keys.c

```diff
@@ -43,25 +65,72 @@ algorand_public_key(const cx_ecfp_private_key_t *privateKey, uint8_t *buf)
   // representing the coordinates are in reverse order.
 
   for (int i = 0; i < 32; i++) {
-    buf[i] = publicKey.W[64-i];
+    public_key_out_buffer[i] = publicKey.W[64-i];
   }
 
   if (publicKey.W[32] & 1) {
-    buf[31] |= 0x80;
+    public_key_out_buffer[31] |= 0x80;
   }
   return 32;
 }
 
-size_t fetch_public_key(uint32_t accountId, uint8_t* pubkey){
-  if(!current_pubkey.initialized ||
-     current_pubkey.accountID != accountId){
-    cx_ecfp_private_key_t privateKey;
-    algorand_key_derive(accountId, &privateKey);
-    algorand_public_key(&privateKey, current_pubkey.pubkey);
-    memset(&privateKey, 0, sizeof(privateKey));
-    current_pubkey.accountID = accountId;
-    current_pubkey.initialized = true;
+
+/*
+* This function returns a public key conresponding to the account ID given.
+* The function fails if the the size of out_pub_key is smaller than 32 bytes (validated with out_pub_key_size arg) .
+*/
+void fetch_public_key(uint32_t account_id, uint8_t* out_pub_key, const uint32_t out_pub_key_size)
+{
+  cx_ecfp_private_key_t private_key;
+  explicit_bzero(&private_key, sizeof(private_key));
+
+  BEGIN_TRY
+  {
+    TRY
+    {
+      algorand_key_derive(account_id, &private_key);
+      get_public_key_from_private_key(&private_key, out_pub_key, out_pub_key_size);
+    }
+    FINALLY
+    {
+      explicit_bzero(&private_key, sizeof(private_key));
+    }
```

Comment:
I'm probably missing something but this try catch is very confusing for me. 
Assuming `algorand_key_derive` failed, you'll have garbage in `out_pub_key` and you won't know about it. Isn't it suppose to throw something if it fails? or do we just suppress the error?